### PR TITLE
BaSP-TG-29: Create and implement global Form component

### PR DESCRIPTION
**Pull request by lucasrendo**
The way the component works is the following:
1. It receives by props all the data necessary for functioning: titles for the labels, id for the name of the properties of the instance, options if there's a list of objects which in turn have an id for referencing in the database and text for showing in the select, etc, input type, etc.
   _Note_: There's a mock implementation of this as example in the index of timesheet, defined in line 75.
2. With those props it maps all necessary inputs using the type property recieved in props to determine if the input must be a select, a date, a checkbox, etc.
   _Note_: in the case of the select, the type is used to render the select and the option elements are mapped from the options property also recieved in props.
3. Once everything is rendered it creates a state that stores all input values and is set to an empty string by default (in the case of all inputs except checkbox which is set to false)
4. From there it listens for changes on the input and a function assign the new values depending on the type of input

The fetch method is set by the calling path. If the path that takes to the form includes an id as parameter in the URL the is set to a PUT request, if the path does not contain an id is set to POST request. The path to go fetch the data is also recieved by the URL by using the router effects.

_NOTE_: The implementation is not done so the fetch, which can be tested after implementation, is not currently working. This includes the related feature of autocompleting the fields on update state

### CODE REVIEW

File: src/Components/Shared/Form/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  font-weight: bolder;
+  cursor: pointer;
+}
+
+.check input {
+  flex: 2;
+  height: 15px;
+  width: 15px;
+  cursor: pointer;
+}
```
Comment (lautaro23) [inline src/Components/Shared/Form/styles.module.css:120]: solo le falta un espacio al final!

**Comment by lucasrendo** [comment]
Added functionality to communicate with the server. Functionalities
- POST: validate that necessary fields are filled before sending and is set to POST method by default when accessing the form
- PUT: accessed through the edit button on the list. It takes to a new route which is 'entity/:id' and set the fetch method to PUT taking the id in the url parameter to tell the server which task to update. It autocompletes all the fields on render and take you back to the list after submit

**Review by FranGutierrez — APPROVED** [review]

**Review by lautaro23 — APPROVED** [review]

**Review by Julian25 — COMMENTED** [review]
Excelente trabajo!

**Review by Julian25 — APPROVED** [review]

File: src/Components/Shared/Form/Form.jsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
+import { useState, useEffect } from 'react';
```
Comment (BaltazarCortina1) [inline src/Components/Shared/Form/Form.jsx:1]: Los archivos deberian ser `index.jsx` e `index.module.css`

File: src/Components/Shared/Form/Form.jsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        return (
+          <div
+            key={item.id}
+            className={item.type === 'checkbox' ? style.check : style.inputContainer}
+          >
+            {item.type !== 'checkbox' && <label htmlFor={item.id}>{item.title}</label>}
+            {item.type === 'select' ? (
+              <select
+                id={item.id}
+                required={item.required && item.required}
```
Comment (BaltazarCortina1) [inline src/Components/Shared/Form/Form.jsx:107]: Aca es necesario poner item.required dos veces?

File: src/Components/Shared/Form/Form.jsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                value={inputValues ? inputValues[item.id] : ''}
+                onChange={(e) => setInputValues({ ...inputValues, [item.id]: e.target.value })}
```
Comment (BaltazarCortina1) [inline src/Components/Shared/Form/Form.jsx:109]: Aca no seria lo mismo pasarle handleChange ?

File: src/Components/TimeSheets/index.js [hunk 5]
```diff
@@ -20,16 +78,71 @@ function TimeSheets() {
       // eslint-disable-next-line no-console
       console.error(error);
     }
-  }, []);
+  };
```
Comment (BaltazarCortina1) [inline src/Components/TimeSheets/index.js:81]: Si aca no le pones el `[]` se va a ejecutar cada vez que cambie algo en el componente

Comment (lucasrendo) [reply src/Components/TimeSheets/index.js:81]: Si, no se porque paso eso, no es la idea hacer eso

**Pull request merged**